mesh/mesh.cpp

```diff
@@ -6194,196 +6195,87 @@ void Mesh::LoadPatchTopo(std::istream &input, Array<int> &edge_to_knot)
    FinalizeTopology();
    CheckBdrElementOrientation(); // check and fix boundary element orientation
 
-   /* Generate knot 2 edge mapping -- if edges are not specified in the mesh file
+   /* Generate edge to knotvector mapping if edges are not specified in the mesh file
       See data/two-squares-nurbs-autoedge.mesh for an example */
-   if (edge_to_knot.Size() == 0)
+   Array<int> ukv_to_pkv;
+   if (edge_to_ukv.Size() == 0)
    {
-      edge_vertex = new Table(NumOfEdges, 2);
-      edge_to_knot.SetSize(NumOfEdges);
-      constexpr int notset = -9999999;
-      edge_to_knot = notset;
-      Array<int> edges;
-      Array<int> oedge;
-      int knot = 0;
-
-      Array<int> edge0, edge1;
-      int flip = 1;
-      if (Dimension() == 2)
-      {
-         edge0.SetSize(2);
-         edge1.SetSize(2);
-
-         edge0[0] = 0; edge1[0] = 2;
-         edge0[1] = 1; edge1[1] = 3;
-         flip = 1;
-      }
-      else if (Dimension() == 3)
-      {
-         edge0.SetSize(9);
-         edge1.SetSize(9);
+      GetEdgeToUniqueKnotvector(edge_to_ukv, ukv_to_pkv);
+      mfem::out << "Generated edge to knot mapping:" << endl;
+      edge_to_ukv.Print(mfem::out);
+   }
+}
 
-         edge0[0] = 0; edge1[0] = 2;
-         edge0[1] = 0; edge1[1] = 4;
-         edge0[2] = 0; edge1[2] = 6;
+void Mesh::GetEdgeToUniqueKnotvector(Array<int> &edge_to_ukv, Array<int> &ukv_to_pkv) const
+{
+   int dim = Dimension();
+   // Local (per-patch) variables
+   int kvidx;
+   Array<int> kvs(dim); // knotvector indices for a patch
+   Array<int> edges, oedges;
+   Array<int> v(2);
+   int d; // local dimension
```

Comment:
A very minor improvement would be defining `const int d = ` inside the loops below.